jupyterhub/templates/hub/secret.yaml

```diff
@@ -16,7 +16,9 @@ data:
   values.yaml: {{ $values | toYaml | b64enc | quote }}
 
   # Used to mount CONFIGPROXY_AUTH_TOKEN on hub/proxy pods for mutual trust
-  proxy.token: {{ .Values.proxy.secretToken | required "Proxy token must be a 32 byte random string generated with `openssl rand -hex 32`!" | b64enc | quote }}
+  JupyterHub.proxy_auth_token: {{ include "jupyterhub.config.JupyterHub.proxy_auth_token" . | required "assert hack 1" | b64enc | quote }}
```

Comment:
I assume these 'required' messages are leftover debug messages?

Reply:
I did not work to provide a user readable formulation of the messages, but I like the idea of having a hard failure during template rendering if this assumption would break for some reason because that would then cause runtime errors hard to debug anyhow.

Do you suggest a change?

Reply:
If it's required, it should have a message saying it's required. It doesn't need to be perfect or verbose, but it shouldn't be an opaque "hack 1" which to me communicates that a TODO item wasn't finished. Even "this shouldn't happen" is more meaningful, as it communicates that it's asserting that a branch has been taken that shouldn't be. Something like "the chart should not allow this value to be undefined" might best communicate what's wrong.

Reply:
I'll update the text! Note though that this is the output of the named template, which will render the explicitly set value, previous value, OR the new generated value.

Reply:
Resolved by b94fec5